go/vt/vttablet/tabletmanager/restore.go

```diff
@@ -83,7 +83,24 @@ func (agent *ActionAgent) restoreDataLocked(ctx context.Context, logger logutil.
 	// Record local metadata values based on the original type.
 	localMetadata := agent.getLocalMetadataValues(originalType)
 	tablet := agent.Tablet()
-	dir := fmt.Sprintf("%v/%v", tablet.Keyspace, tablet.Shard)
+
+	keyspace := tablet.Keyspace
+	keyspaceInfo, err := agent.TopoServer.GetKeyspace(ctx, keyspace)
+	if err != nil {
+		return err
+	}
+	// For a SNAPSHOT keyspace, we have to look for backups of BaseKeyspace
+	// so we will pass the BaseKeyspace in RestoreParams instead of tablet.Keyspace
+	// Keyspace creation checks that BaseKeyspace is set for a SNAPSHOT keyspace,
+	// so we don't need to check again here
+	if keyspaceInfo.KeyspaceType == topodatapb.KeyspaceType_SNAPSHOT {
+		keyspace = keyspaceInfo.BaseKeyspace
```

Comment:
Maybe add a log line that we are overriding the keyspace used for restore because it's a snapshot keyspace with given base keyspace.

Reply:
done